scripts/test_matrix.py

```diff
@@ -0,0 +1,184 @@
+]
+
+neg_unit = [
+	['dgram g00n13s', prov_dgram],
+	['rdm g00n13s' 	, prov_rdm],
+	['msg g00n13s'	, prov_msg],
+]
+
+simple = [
+	['cm_data', 			prov_msg],
+	# cq_data test supports only FI_EP_MSG
+	['cq_data', 			{'ofi-rxm': 0, 'psm': 0, 'psm2': 0, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	['dgram', 			prov_dgram],
+	['dgram_waitset', 		prov_dgram],
+	['msg', 			prov_msg],
+	['msg_epoll', 			prov_msg],
+	['msg_sockets', 		prov_msg],
+	['poll -t queue', 		{'ofi-rxm': 1, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['poll -t counter',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm',				prov_rdm],
+	# rdm_rma_simple requires FI_RMA_EVENT
+	['rdm_rma_simple',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_rma_trigger',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg"),			{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg","notx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("msg","norx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm"),			{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm","notx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("rdm","norx"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram"),		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram","notx"),	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	[sh_ctx("dgram","norx"),	{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['rdm_tagged_peek',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 1}],
+	['scalable_ep',			{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+	['cmatose',			prov_msg],
+	['rdm_shared_av',		{'ofi-rxm': 0, 'psm': 1, 'psm2': 1, 'sockets': 1, 'udp': 0, 'verbs': 0}],
+]
+
+standard = [
+	['msg_pingpong',		prov_msg],
+	['msg_pingpong -v',		prov_msg],
+	['msg_pingpong -k',		prov_msg],
+	['msg_pingpong -k -v',		prov_msg],
+	['msg_bw',			prov_msg],
+	['msg_bw -v',			prov_msg],
+	['msg_bw -k',			prov_msg],
+	['msg_bw -k -v',		prov_msg],
+	['rdm_pingpong',		prov_rdm],
+	['rdm_pingpong -v',		prov_rdm],
+	['rdm_pingpong -k',		prov_rdm],
+	['rdm_pingpong -k -v',		prov_rdm],
+	['rdm_tagged_pingpong',		prov_rdm],
+	['rdm_tagged_pingpong -v',	prov_rdm],
+	['rdm_tagged_pingpong -k',	prov_rdm],
+	['rdm_tagged_pingpong -k -v',	prov_rdm],
+	['rdm_tagged_bw',		prov_rdm],
+	['rdm_tagged_bw -v',		prov_rdm],
+	['rdm_tagged_bw -k',		prov_rdm],
+	['rdm_tagged_bw -k -v',		prov_rdm],
+	['dgram_pingpong',		prov_dgram],
+	['dgram_pingpong -v',		prov_dgram],
+	['dgram_pingpong -k',		prov_dgram],
+	['dgram_pingpong -k -v',	prov_dgram],
```

Comment:
psm and psm2 support dgram, but dgram_pinpong is not supported because of FI_CONTEXT mode.